src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp

```diff
@@ -52,6 +52,112 @@ bool SplineC2C<ST>::write_splines(hdf_archive& h5f)
+  so if we put rot_mat at top left corner of tmpU, then we can apply tmpU to the
+  coefs safely regardless of padding.
+
+  NB: For splines (typically) BasisSetSize >> OrbitalSetSize, so the spl_coefs
+  "matrix" is very tall and skinny.
+*/
+template<typename ST>
+void SplineC2C<ST>::applyRotation(const ValueMatrix& rot_mat, bool use_stored_copy)
+{
+  // SplineInst is a MultiBspline. See src/spline2/MultiBspline.hpp
+  const auto spline_ptr = SplineInst->getSplinePtr();
+  assert(spline_ptr != nullptr);
+  const auto spl_coefs      = spline_ptr->coefs;
+  const auto Nsplines       = spline_ptr->num_splines;
+  const auto coefs_tot_size = spline_ptr->coefs_size;
+  const auto BasisSetSize   = coefs_tot_size / Nsplines;
+  const auto TrueNOrbs      = rot_mat.size1(); // == Nsplines - padding
+  assert(OrbitalSetSize >= TrueNOrbs);
+
+  // Fill top left corner of tmpU with rot_mat
+  ValueMatrix tmpU;
+  tmpU.resize(Nsplines, Nsplines);
+  std::fill(tmpU.begin(), tmpU.end(), 0.0);
+  for (auto i = 0; i < rot_mat.size1(); i++)
+  {
+    for (auto j = 0; j < rot_mat.size2(); j++)
+    {
+      tmpU[i][j] = rot_mat[i][j];
+    }
+  }
+
+  // Apply rotation the dumb way b/c I can't get BLAS::gemm to work...
+  // For SplineC2C we have to handle complex spl_coefs layout by hand.
+  // Each coef is stored as two realtypes adjacent in memory. For
+  // N spline coefficients the spl_coefs array for a SplineC2C object has
+  // length = 2N to account for real/imag terms. A comparison of the layouts
+  // is given below.
+  //
+  // SplineR2R layout:
+  // |====================|
+  // | c0 | c1 | ... | cN |
+  // |====================|
+  // <- N coefs length N ->
+  //
+  // SplineC2C layout:
+  // |===========================================================|
+  // | Re{c0} | Im{c0} | Re{c1} | Im{c1} | ... | Re{cN} | Im{cN} |
+  // |===========================================================|
+  // <-------------------- N coefs length 2N -------------------->
+  std::vector<ST> new_coefs(coefs_tot_size, 0);
+  ST z1{0.};
+  ST z2{0.};
+  ST w1{0.};
+  ST w2{0.};
+  ST newval_real{0.};
+  ST newval_imag{0.};
+  for (auto i = 0; i < BasisSetSize; i++)
+  {
+    for (auto j = 0; j < Nsplines; j++)
+    {
+      const auto cur_elem = 2 * (Nsplines / 2 * i + j);
+      newval_real         = 0.;
+      newval_imag         = 0.;
+      for (auto k = 0; k < Nsplines; k++)
+      {
+        const auto index = 2 * (i * Nsplines / 2 + k);
+        z1               = *(spl_coefs + index);
+        z2               = *(spl_coefs + index + 1);
+        w1               = tmpU[k][j].real();
+        w2               = tmpU[k][j].imag();
+        newval_real += z1 * w1 - z2 * w2;
+        newval_imag += z1 * w2 + z2 * w1;
+      }
+      new_coefs[cur_elem]     = newval_real;
+      new_coefs[cur_elem + 1] = newval_imag;
+    }
+  }
+
+  // Update the coefs
+  std::copy(new_coefs.begin(), new_coefs.end(), spl_coefs);
+
+  /*
```

Comment:
Please remove this commented-out block.  It's okay for it to stay in the original SplineR2R location, but it should be removed here.

Reply:
done.